connect/api/src/main/java/org/apache/kafka/connect/rest/ConnectRestExtension.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.connect.rest;
+
+import org.apache.kafka.common.Configurable;
+import org.apache.kafka.connect.components.Versioned;
+import org.apache.kafka.connect.health.ConnectClusterState;
+
+import java.io.Closeable;
+import java.util.Map;
+
+/**
+ * A plugin interface to allow registration of new JAX-RS resources like Filters, REST endpoints,
+ * providers, etc. The implementations will be discovered by the standard Java {@link
+ * java.util.ServiceLoader} mechanism by  Connect's plugin class loading mechanism.
+ * The implementation class must be packaged in a JAR that includes the
+ * {@code META-INF/services/org.apache.kafka.connect.rest.extension.ConnectRestExtension} containing
+ * the fully qualified name of the implementation class.
+ *
+ * <p> When Connect's worker configuration uses the REST extension implementation class,
+ * upon startup Connect will instantiate the implementation and pass the configuration
+ * to the instance via {@link Configurable#configure(Map)}.
```

Comment:
Nit:
```
<p>When the Connect worker process starts up, it will read its configuration and instantiate all of the
REST extension implementation classes that are specified in the `rest.extension.classes` configuration property.
Connect will then pass its configuration to each extension via the {@link Configurable#configure(Map)} method,
and will then call {@link #register} with a provided context.

<p>When the Connect worker shuts down, it will call the extension's {@link #close} method to allow the
implementation to release all of its resources.
```